neonvm/tools/vm-builder/files/set-disk-quota.sh

```diff
@@ -0,0 +1,21 @@
+#!/neonvm/bin/sh
+
+set -euo pipefail
+
+USAGE="USAGE: /neonvm/bin/set-disk-quota <NEW_SIZE_IN_KB> <MOUNTPOINT>"
+
+if [ "$#" -eq 2 ]; then
+    size="$1"
+    mountpoint="$2"
+else
+    echo "expected 2 args, got $#" >&2
+    echo "$USAGE" >&2
+    exit 1
+fi
+
+echo "Setting disk quota for $mountpoint to $size KB"
+/neonvm/bin/setquota -P 0 0 "$size" 0 0 "$mountpoint"
+
+echo "DONE! Disk quota set"
+/neonvm/bin/repquota -P -a
```

Comment:
This is not needed, it just shows current status. Probably useful for debug.

I'm not 100% we should leave it here, what do you all think?

Reply:
I think we should leave it, if the command doesn't take any noticeable time to run.

Reply:
IMO it should either have more context (e.g., "disk quota set to X") or be removed — otherwise people will be confused by it in the compute logs :sweat_smile:

Reply:
Example of repquota output:

```
root@vm-example-vxc8h:/var/db/postgres/compute# /neonvm/bin/repquota -P -a
*** Report for project quotas on device /dev/vdd
Block grace time: 7days; Inode grace time: 7days
                        Block limits                File limits
Project         used    soft    hard  grace    used  soft  hard  grace
----------------------------------------------------------------------
#0        --   25164       0       0              3     0     0       
```

Reply:
From the e2e test run:

```
*** Report for project quotas on device /dev/vdd
Block grace time: 7days; Inode grace time: 7days
                        Block limits                File limits
Project         used    soft    hard  grace    used  soft  hard  grace
----------------------------------------------------------------------
#0        --      20       0 1048576              2     0     0       
```

at first glance, I think this is kind of hard to read, and we could probably make do with much smaller output (e.g, just showing block used/hard limit and file used/hard limit).

Any ideas if that's possible?

Reply:
I don't think it's possible to make it much better. `repquota` has options to change display format to csv/xml, but then you will need to parse it to display something nice.

This is the full help output:
```
root@vm-example-zb8g4:~# /neonvm/bin/repquota -h
repquota: Utility for reporting quotas.
Usage:
repquota [-vugsi] [-c|C] [-t|n] [-F quotaformat] [-O (default | xml | csv)] (-a | mntpoint)

-v, --verbose                 display also users/groups without any usage
-u, --user                    display information about users
-g, --group                   display information about groups
-P, --project                 display information about projects
-s, --human-readable[=units]  display numbers in human friendly units (MB, GB,
                              ...). Units can be also specified explicitely by
                              an optional argument in format [kgt],[kgt] where
                              the first character specifies space units and the
                              second character specifies inode units
-t, --truncate-names          truncate names to 9 characters
-p, --raw-grace               print grace time in seconds since epoch
-n, --no-names                do not translate uid/gid to name
-i, --no-autofs               avoid autofs mountpoints
-c, --cache                   translate big number of ids at once
-C, --no-cache                translate ids one by one
-F, --format=formatname       report information for specific format
-O, --output=format           format output as xml or csv
-a, --all                     report information for all mount points with
                              quotas
-h, --help                    display this help message and exit
-V, --version                 display version information and exit

Bugs to jack@suse.cz
```

Reply:
Ok. Let's leave it for now, and then maybe remove it later once things are proven stable?

Reply:
Yes, sounds good. I can also remove it right now as I have no strong opinion on this, do you think it'll be better?

Reply:
I think the additional observability is valuable enough in the short term that we should include it for now.